src/test/scala/com/advancedtelematic/campaigner/http/CampaignResourceSpec.scala

```diff
@@ -354,4 +355,40 @@ class CampaignResourceSpec
     }
   }
 
+  "GET /campaigns/:id/stats" should "return correct statistics" in {
+    val campaigns = Campaigns()
+    val groupId = genGroupId.generate
+
+    val successDevices = Gen.listOf(genDeviceId).generate
```

Comment:
Why are we generating a single sample?

Reply:
I'm not sure I understood you. Is your question "why do we only test one campaign case and not make it into a property-based test instead"? If so, then there is no good reason for it indeed and I can gladly convert it into a property-based test.

Reply:
yes, that is what i mean. Let's do it!